File: packages/whisker-hello-element/src/ios/WhiskerHelloElement.swift
// Whisker native element on iOS with local tag `Hello`. The Lynx
// registration string is `whisker-hello-element:Hello` — the
// SwiftPM build plugin prepends `context.package.displayName`
// (the cargo crate name) as the namespace so two unrelated
// packages can both declare a `Hello` element without colliding
// in Lynx's behaviour registry. Phase 7-Φ.H.2.
//
// Demonstrates the Whisker-only API surface (no `import Lynx`,
// no `LynxUI` mentions in author code).
//
// `WhiskerUI` / `WhiskerContext` are typealiases provided by
// WhiskerRuntime that resolve to the underlying Lynx types at
// the Swift type-system level. Stack traces / debugger views
// still show the real `LynxUI` class name — typealiases are a
// source-level concept only.

import UIKit
import WhiskerElements
import WhiskerRuntime

@WhiskerElement("Hello")
@objc(WhiskerHelloElement)
public final class WhiskerHelloElement: WhiskerUI<UIView> {
    @objc public override func createView() -> UIView {
        let v = UIView()
        v.backgroundColor = .systemPink
        return v
    }
}

File: packages/whisker-hello-element/src/android/WhiskerHelloElement.kt
// `<x-hello>` Whisker native element on Android — Phase 7-Φ.H.1
// migration target. Demonstrates the Whisker-only API surface
// (no `com.lynx.tasm.*` imports in author code).
//
// `WhiskerUI` / `WhiskerContext` are typealiases provided by
// `rs.whisker.runtime` that resolve to the underlying Lynx types
// at Kotlin's type-system level. Stack traces / debugger views
// still show the real `LynxUI` class names — typealiases are a
// source-level concept only.

package rs.whisker.elements.hello

import android.content.Context
import android.graphics.Color
import android.view.View
import rs.whisker.annotations.WhiskerElement
import rs.whisker.runtime.WhiskerContext
import rs.whisker.runtime.WhiskerUI

@WhiskerElement("x-hello")
open class WhiskerHelloElement(context: WhiskerContext) : WhiskerUI<View>(context) {

    override fun createView(context: Context): View {
        val view = View(context)
        view.setBackgroundColor(Color.YELLOW)
        return view
    }
}
